source/kraken/maintenance_worker.cpp

```diff
@@ -485,29 +485,45 @@ void MaintenanceWorker::handle_rt_in_batch(const std::vector<AmqpClient::Envelop
     }
 }
 
-std::vector<AmqpClient::Envelope::ptr_t> MaintenanceWorker::consume_in_batch(const std::string& consume_tag,
+std::vector<AmqpClient::Envelope::ptr_t> MaintenanceWorker::consume_in_batch(const std::string& queue_name,
+                                                                             const std::string& consume_tag,
                                                                              size_t max_nb,
-                                                                             size_t timeout_ms,
                                                                              bool no_ack) {
     assert(consume_tag != "");
     assert(max_nb);
-
     std::vector<AmqpClient::Envelope::ptr_t> envelopes;
-    envelopes.reserve(max_nb);
-    size_t consumed_nb = 0;
+
+    // count ready messages (prefetched messages are excluded from this count)
+    auto count_messages_consumers = declare_queue_with_counts(queue_name);
```

Comment:
```suggestion
    // declare_queue_with_counts is called but no queues are actually created
    auto count_messages_consumers = declare_queue_with_counts(queue_name);
```

Reply:
Done in https://github.com/hove-io/navitia/pull/3968/commits/868fc7b3fe6b5012a39fd6e4ae3da3d0e85a022f